File: codingjs/hasOne.js
function hasOne(n: number): number {
  return hasOneHelper(n);
}
function hasOneHelper(n: number): number {
  if (n < 10 && n != 1) return false;
  if (n % 10 === 1) {
    return true;
  } else {
    return hasOneHelper(Math.floor(n / 10));
  }
}

File: codingbat/java/hasOne.java
public boolean hasOne(int n) {
  return hasOneHelper(n);
}

public boolean hasOneHelper(int n) {
  // base case
  if (n < 10) return n == 1;
  
  if (n % 10 == 1) return true;
  n /= 10;
  return hasOneHelper(n);
}
